Gems/Atom/Tools/AtomToolsFramework/Code/Source/Util/Util.cpp

```diff
@@ -661,6 +662,197 @@ namespace AtomToolsFramework
+
+        return GetPathsForAssetSourceDependencies(sourceInfo);
+    }
+
+    AZStd::vector<AZStd::string> GetPathsForAssetSourceDependenciesByPath(const AZStd::string& sourcPath)
+    {
+        bool found = false;
+        AZ::Data::AssetInfo sourceInfo;
+        AZStd::string watchFolder;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            found,
+            &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath,
+            GetPathWithoutAlias(sourcPath).c_str(),
+            sourceInfo,
+            watchFolder);
+
+        return GetPathsForAssetSourceDependencies(sourceInfo);
+    }
+
+    AZStd::vector<AZStd::string> GetPathsForAssetSourceDependents(const AZ::Data::AssetInfo& sourceInfo)
+    {
+        AzToolsFramework::AssetDatabase::AssetDatabaseConnection assetDatabaseConnection;
+        assetDatabaseConnection.OpenDatabase();
+
+        AzToolsFramework::AssetDatabase::SourceDatabaseEntry sourceEntry;
+        assetDatabaseConnection.QuerySourceBySourceName(
+            sourceInfo.m_relativePath.c_str(),
+            [&sourceEntry](const AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
+            {
+                sourceEntry = entry;
+                return false;
+            });
+
+        if (sourceEntry.m_sourceGuid.IsNull())
+        {
+            return {};
+        }
+
+        AZStd::string scanFolderPath;
+        assetDatabaseConnection.QueryScanFolderByScanFolderID(
+            sourceEntry.m_scanFolderPK,
+            [&scanFolderPath](const AzToolsFramework::AssetDatabase::ScanFolderDatabaseEntry& entry)
+            {
+                scanFolderPath = entry.m_scanFolder;
+                return false;
+            });
+
+        auto absolutePath = AZ::IO::Path(scanFolderPath) / sourceEntry.m_sourceName;
+
+        AZStd::vector<AZStd::string> sourcePaths;
+        assetDatabaseConnection.QuerySourceDependencyByDependsOnSource(
+            sourceEntry.m_sourceGuid,
+            sourceEntry.m_sourceName.c_str(),
+            absolutePath.FixedMaxPathStringAsPosix().c_str(),
+            AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::DEP_Any,
+            [&sourcePaths, &assetDatabaseConnection](AzToolsFramework::AssetDatabase::SourceFileDependencyEntry& entry)
+            {
+                AZStd::string sourceName;
+                assetDatabaseConnection.QuerySourceBySourceGuid(
+                    entry.m_sourceGuid,
+                    [&sourceName](AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
+                    {
+                        sourceName = entry.m_sourceName;
+                        return false;
+                    });
+
+                if (!sourceName.empty())
+                {
+                    sourcePaths.emplace_back(GetAbsolutePathForSourceAsset(sourceName));
+                }
+                return true;
```

Comment:
Shouldn't we check for duplicates in the sourcePaths before emplacing back.

Reply:
Addressed by sorting the container and removing nonunique values